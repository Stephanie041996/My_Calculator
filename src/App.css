body {
  display: flex;
  justify-content: center;
  align-items: center;
}

button {
  height: 100px;
  border-radius: 0;
  border: 1px solid gray;
  text-align: center;
  color: black;

}

.display-result {
  width: 700px;
  height: 100px;
  margin-left: auto;
  margin-right: auto;
}

#result {
  width: 695px;
  height: 100px;
  border: 1px solid gray;
  background-color: gray;
  color: white;
  font-weight: bold;
 text-align: right;
}

.button-panel {
  width: 700px;
  margin-left: auto;
  margin-right: auto;
}
