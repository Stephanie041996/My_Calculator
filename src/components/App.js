import React, { Component } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import Calculate from '../logic/calculate';
import '../styles/App.css';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    };
  }

 handleClick = (buttonName) => {
   const calculation = Calculate(buttonName, this.state);
   this.state({
     total: calculation.total,
     next: calculation.next,
     operation: calculation.operation,
   });
 }

   render = () => {
     const { next, total } = this.state;
     return (
       <>
         <h1>Calculator App</h1>
         <Display Calculate={next || total} />
         <ButtonPanel clickHandler={(e) => this.handleClick(e.target.innerText)} />
       </>
     );
   }
}

export default App;
