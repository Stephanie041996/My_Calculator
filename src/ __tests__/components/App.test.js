import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../components/App';

describe('App', () => {
  it('should render calculator', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
