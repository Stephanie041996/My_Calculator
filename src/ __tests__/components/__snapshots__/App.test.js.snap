// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render calculator 1`] = `
Array [
  <h1>
    Calculator App
  </h1>,
  <div
    className="display-result"
  >
    <input
      defaultValue={null}
      id="result"
    />
  </div>,
  <div
    className="button-panel"
  >
    <div>
      <button
        name="AC"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        name="+/-"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        name="%"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        name="รท"
        onClick={[Function]}
        style={
          Object {
            "background": "color",
            "width": "50%",
          }
        }
        type="button"
      >
        รท
      </button>
    </div>
    <div>
      <button
        name="7"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        name="8"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        name="9"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        name="X"
        onClick={[Function]}
        style={
          Object {
            "background": "color",
            "width": "50%",
          }
        }
        type="button"
      >
        X
      </button>
    </div>
    <div>
      <button
        name="4"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        name="5"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        name="6"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        name="-"
        onClick={[Function]}
        style={
          Object {
            "background": "color",
            "width": "50%",
          }
        }
        type="button"
      >
        -
      </button>
    </div>
    <div>
      <button
        name="1"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        name="2"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        name="3"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        name="+"
        onClick={[Function]}
        style={
          Object {
            "background": "color",
            "width": "50%",
          }
        }
        type="button"
      >
        +
      </button>
    </div>
    <div>
      <button
        name="0"
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        name="."
        onClick={[Function]}
        style={
          Object {
            "background": "white",
            "width": "50%",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        name="="
        onClick={[Function]}
        style={
          Object {
            "background": "color",
            "width": "50%",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>,
]
`;
